### DISCUSSION TITLE
Good choice for my use case?

### DISCUSSION THREAD

**Discussion by michaelc95** [trapexit/mergerfs#917] [General]
Just discovered your project. Such a great idea. I think my use case is prefect for it..
I have a directory structure as follows

/Plots
          /plot1 = 8tb xfs mount
          /plot2  = 16 tb xfs mount
          /plot3  = 12 tb xfs mount
          /plot4  - rest are just an empty directories for mounting future drives
          ....
          /plot24 

My goal is to evenly distribute 101 GB files across all mounted drives so like a /Plots-all directory
Each file is unique and written once and basically only gets read from then on. What would Mergerfs's reaction be if I did /plots/plot* as the glob location? Would it try to stick files in the directories that don't have mounted filesystems? 

How could I add on drives to the mergerfs on the fly? I
Is it just a matter of mounting it up under the glob location and it will just start using it? 
Is there any significant performance impact? 
What would be the best options to list for my use case? I dont understand what half of them do. 

Thanks for putting together this cool FS and sharing it with the world. 
- Mike

**Comment by trapexit** [comment]
Heh. You're doing Chia farming and want to know if it will work? 

I've not dabbled in it myself but from what I gather mergerfs has gotten some attention by Chia farmers and I've not heard of any particular problems.

>What would Mergerfs's reaction be if I did /plots/plot* as the glob location? Would it try to stick files in the directories that don't have mounted filesystems?

Everything that matches the glob at mount time would be added as a branch. I don't believe I filter out anything based on the type (dir vs regular vs etc.) Generally speaking it's best to have your mounts in a single location and just use '/mymounts/*'. For instance I just keep a directory in /etc/mergerfs/pools with symlinks to the relevant branches I want. Then have my branches argument be '/etc/mergerfs/pools/main/*'

>How could I add on drives to the mergerfs on the fly?

https://github.com/trapexit/mergerfs#mergerfs-pseudo-file-deprecated

> Is it just a matter of mounting it up under the glob location and it will just start using it?

Yes? I don't understand the question really.

> Is there any significant performance impact?

Depends entirely on the workload. See the performance and benchmarking sections of the docs.

> What would be the best options to list for my use case? I dont understand what half of them do.

I don't know. Most things are preference. If you don't understand them then you probably don't need to worry about it. Start with the suggested options as shown in the docs and move from there.
